src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -189,8 +189,11 @@ public Task SaveCache()
             {
                 using (var writer = new BinaryWriter(stream))
                 {
-                    writer.Write(_commitCache.Count);
-                    foreach (var (file, value) in _commitCache)
+                    // Create a snapshot of the commit cache to avoid writing corrupected data
+                    var commitCache = _commitCache.ToList();
```

Comment:
any reason here why ToList can resolve this problem?

Reply:
after `writer.Write(_commitCache.Count)` and before `foreach (xxx in _commitCache)` is completed, there is a tiny chance that a new entry is added to `_commitCache`, causing the count on disk to not match the items count.

Reply:
so we should lock _commitCache right?

Reply:
I looked the code, didn't fond the tiny chance to change _commitCache during save unless we have some bugs, so we should take a look at the root cause